Network/Socket/ByteString/IO.hsc

```diff
@@ -310,8 +311,8 @@ recvMsg :: Socket  -- ^ Socket
         -> IO (SockAddr, ByteString, [Cmsg], MsgFlag) -- ^ Source address, received data, control messages and message flags
 recvMsg s siz clen flags = do
     bs <- create siz $ \ptr -> zeroMemory ptr (fromIntegral siz)
-    bufsiz <- getBufsiz bs
-    (addr,len,cmsgs,flags') <- recvBufMsg s [bufsiz] clen flags
-    let bs' | len < siz = let PS buf 0 _ = bs in PS buf 0 len
-            | otherwise = bs
-    return (addr, bs', cmsgs, flags')
+    withBufSizs [bs] $ \bufsizs -> do
```

Comment:
I would specialize this case by implementing an `withBufsiz` that simply wraps just a single `ByteString` and avoids the loop, ...

Reply:
`withForeignPtr` is used directly now.

Reply:
Thanks this is resolved.